src/components/WeChatQRLogin.tsx

```diff
@@ -11,29 +11,56 @@ declare global {
 export default function WeChatQRLogin({ appid }: { appid: string }) {
   const containerRef = useRef<HTMLDivElement>(null);
   const router = useRouter();
-  const callbackUrl = router.query.callbackUrl as string || '/';
+  const [isLoading, setIsLoading] = useState(true);
+
+  // 获取当前路径作为回调URL
+  const currentPath = typeof window !== 'undefined'
+    ? window.location.pathname + window.location.search
+    : '/';
+  const callbackUrl = router.query.callbackUrl as string ||
+    (currentPath !== '/auth/login' ? currentPath : '/');
```

Comment:
Can you pass in the callbackUrl from the caller (ie. add a callbackUrl parameter to the WeChatQRLogin() function, instead of recomputing it here?